src/structures_model.jl

```diff
@@ -5,65 +5,68 @@ abstract type EnergyModel end
 Operational Energy Model without investments.
 
 # Fields
-- **`emission_limit`** is a dictionary with individual emission limits as `TimeProfile` for \
-each emission resource `ResourceEmit`.\n
-- **`emission_price::Dict{ResourceEmit, TimeProfile}`** are the prices for the different \
-emissions types considered.\n
+- **`emission_limit::Dict{<:ResourceEmit, <:TimeProfile}`** is a dictionary with \
+individual emission limits as `TimeProfile` for each emission resource `ResourceEmit`.\n
+- **`emission_price::Dict{<:ResourceEmit, <:TimeProfile}`** are the prices for the \
+different emissions types considered.\n
 - **`co2_instance`** is a `ResourceEmit` and corresponds to the type used for CO₂.\n
 """
 struct OperationalModel <: EnergyModel
-    emission_limit::Dict{ResourceEmit, TimeProfile}
-    emission_price::Dict{ResourceEmit, TimeProfile}
+    emission_limit::Dict{<:ResourceEmit, <:TimeProfile}
+    emission_price::Dict{<:ResourceEmit, <:TimeProfile}
```

Comment:
Since `ResourceEmit` is a composite type, I guess `<:` can be dropped for that type?

Reply:
No, but a parametric type. Hence, it is required.
`Dict(CO2 => FixedProfile(10)) isa Dict{ResourceEmit, <:TimeProfile` would return false.